### PR TITLE
Parallelize job gcs bucket processing

### PR DESCRIPTION
**Pull request by stbenjam**
[TRT-1190](https://issues.redhat.com//browse/TRT-1190)

We're spending a lot of time processing gcs buckets and importing jobs, such that when combined with the matview refreshes we're regularly approaching or exceeding the 60 minute window. This changes the gcs importer code to import up to 10 jobs in parallel.  Gorm is (mostly) thread safe, and I think I've protected all the places in our code that isn't.

### CODE REVIEW

File: pkg/prowloader/prow.go [hunk 1]
```diff
@@ -153,46 +168,103 @@ func (pl *ProwLoader) LoadProwJobsToDB() []error {
...
+	// Start pl.maxConcurrency consumers
+	var wg sync.WaitGroup
+	for i := 0; i < pl.maxConcurrency; i++ {
+		wg.Add(1)
+		go func(ctx context.Context) {
+			defer wg.Done()
+			for job := range queue {
+				if ctx.Err() != nil {
+					errsCh <- ctx.Err()
+					fmt.Println("consumer exiting, got error")
```
Comment (dgoodwin) [inline pkg/prowloader/prow.go:187]: Lets flip to logrus logging WithError

File: pkg/prowloader/prow.go [hunk 2]
```diff
@@ -484,15 +561,24 @@ func (pl *ProwLoader) prowJobToJobRun(pj prow.ProwJob, release string, newJobRun
 			}
 		}
 		if saveDB {
-			pl.dbc.DB.Save(&dbProwJob)
+			fmt.Println("update")
```
Comment (dgoodwin) [inline pkg/prowloader/prow.go:564]: logrus or remove

File: pkg/prowloader/prow.go [hunk 3]
```diff
@@ -623,30 +711,44 @@ func (pl *ProwLoader) findOrAddPullRequests(refs *prow.Refs, pjPath string) []mo
...
 
+	pl.suiteCacheLock.RLock()
 	if id, ok := pl.suiteCache[name]; ok {
+		pl.suiteCacheLock.RUnlock()
 		return id
 	}
+	pl.suiteCacheLock.RUnlock()
 
+	pl.suiteCacheLock.Lock()
+	defer pl.suiteCacheLock.Unlock()
```
Comment (dgoodwin) [inline pkg/prowloader/prow.go:751]: Looks like an unnecessary lock/unlock here.

**Review by dgoodwin — COMMENTED** [review]
It looks ok to me, just a couple nits. I'm going to hold just for a bit until we settle things down in the db world.

/hold

File: pkg/prowloader/prow.go (hunk 3 above)
Comment (stbenjam) [reply pkg/prowloader/prow.go:750]: It's needed since we're manipulating the cache a little bit further down

**Comment by stbenjam** [comment]
> It looks ok to me, just a couple nits. I'm going to hold just for a bit until we settle things down in the db world.

Yea sounds good, I also need to dig into why e2e is failing. Sync completes fine, but tests are failing

**Comment by stbenjam** [comment]
/hold cancel

Working now

**Pull request merged**